cpu/esp32/periph/can.c

```diff
@@ -163,26 +140,26 @@ static void _esp_can_isr(candev_t *candev)
     assert(dev->candev.event_callback);
 
     if (dev->events & ESP_CAN_EVENT_BUS_OFF) {
-        dev->events &= ~ESP_CAN_EVENT_BUS_OFF;
+        dev->events ^= ESP_CAN_EVENT_BUS_OFF;
         dev->candev.event_callback(&dev->candev, CANDEV_EVENT_BUS_OFF, NULL);
     }
 
     if (dev->events & ESP_CAN_EVENT_ERROR_PASSIVE) {
-        dev->events &= ~ESP_CAN_EVENT_ERROR_PASSIVE;
+        dev->events ^= ESP_CAN_EVENT_ERROR_PASSIVE;
         dev->candev.event_callback(&dev->candev,
                                    CANDEV_EVENT_ERROR_PASSIVE, NULL);
     }
 
     if (dev->events & ESP_CAN_EVENT_ERROR_WARNING) {
-        dev->events &= ~ESP_CAN_EVENT_ERROR_WARNING;
+        dev->events ^= ESP_CAN_EVENT_ERROR_WARNING;
         dev->candev.event_callback(&dev->candev,
                                    CANDEV_EVENT_ERROR_WARNING, NULL);
     }
 
     if (dev->events & ESP_CAN_EVENT_TX_ERROR) {
         /* a pending TX confirmation has also to be deleted on TX bus error */
-        dev->events &= ~ESP_CAN_EVENT_TX_ERROR;
-        dev->events &= ~ESP_CAN_EVENT_TX_CONFIRMATION;
+        dev->events ^= ESP_CAN_EVENT_TX_ERROR;
+        dev->events ^= ESP_CAN_EVENT_TX_CONFIRMATION;
```

Comment:
Here the change could indeed be a problem if the bit is not set, which is not guaranteed by the condition before.